dev/PushNotifications/PushNotificationsLongRunningTask/NotificationListener.cpp

```diff
@@ -36,39 +36,30 @@ HRESULT NotificationListener::RuntimeClassInitialize(
 }
 CATCH_RETURN();
 
-STDMETHODIMP_(HRESULT __stdcall) NotificationListener::OnRawNotificationReceived(unsigned int payloadLength, _In_ byte* payload, _In_ HSTRING correlationVector) noexcept try
+STDMETHODIMP_(HRESULT __stdcall) NotificationListener::OnRawNotificationReceived(
+    unsigned int payloadLength,
+    _In_ byte* payload,
+    _In_ HSTRING correlationVector) noexcept try
 {
-    HRESULT hr{ S_OK };
+    auto logTelemetry = PushNotificationLongRunningTaskTelemetry::OnRawNotificationReceived::Start(correlationVector);
+    auto lock = m_lock.lock_exclusive();
 
-    auto logTelemetry{ wil::scope_exit([&]() {
-        PushNotificationLongRunningTaskTelemetry::LogOnRawNotificationReceived(hr, WindowsGetStringRawBuffer(correlationVector, nullptr));
-    }) };
+    winrt::com_array<uint8_t> payloadArray{ payload, payload + (payloadLength * sizeof(uint8_t)) };
 
-    try
+    if (!m_foregroundSinkManager->InvokeForegroundHandlers(m_appId, payloadArray, correlationVector, payloadLength))
     {
-        auto lock = m_lock.lock_exclusive();
-
-        winrt::com_array<uint8_t> payloadArray{ payload, payload + (payloadLength * sizeof(uint8_t)) };
-
-        if (!m_foregroundSinkManager->InvokeForegroundHandlers(m_appId, payloadArray, payloadLength))
+        if (m_comServerClsid == winrt::guid())
         {
-            if (m_comServerClsid == winrt::guid())
-            {
-                THROW_IF_FAILED(PushNotificationHelpers::ProtocolLaunchHelper(m_processName, payloadLength, payload));
-            }
-            else
-            {
-                THROW_IF_FAILED(PushNotificationHelpers::PackagedAppLauncherByClsid(m_comServerClsid, payloadLength, payload));
-            }
+            THROW_IF_FAILED(PushNotificationHelpers::ProtocolLaunchHelper(m_processName, payloadLength, payload));
+        }
+        else
+        {
+            THROW_IF_FAILED(PushNotificationHelpers::PackagedAppLauncherByClsid(m_comServerClsid, payloadLength, payload));
         }
-
-        return hr;
-    }
-    catch (...)
-    {
-        hr = wil::ResultFromCaughtException();
-        throw;
     }
+
+    logTelemetry.Stop();
```

Comment:
Can we use scope_exit here instead for stop telemetry?

Reply:
Good catch!